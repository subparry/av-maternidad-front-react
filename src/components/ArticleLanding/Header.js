/* eslint-disable jsx-a11y/img-redundant-alt */
import React from "react";
import ArticleAuthorCard from "./ArticleAuthorCard";

const Header = ({ title, subtitle, imageUrl, estimatedTimeToRead }) => {
  return (
    <section className="article-header">
      <h1 className="article-title">{title}</h1>
      {subtitle && <h3>{subtitle}</h3>}
      <ArticleAuthorCard estimatedTimeToRead={estimatedTimeToRead} />
      <img
        src={imageUrl}
        alt="article main image"
        className="article-main-image"
      />
    </section>
  );
};

export default Header;
