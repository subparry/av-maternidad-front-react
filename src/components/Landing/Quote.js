import React from "react";
import "./styles/quote.scss";
const Quote = () => {
  return (
    <section data-testid="quote-section" className="quote-section">
      <p className="parenthood-quote text-md">
        -La funci√≥n de un hijo es vivir su propia vida, no vivir la vida que su
        padre ansioso cree que debe vivir. <em>-A.S. Neill.</em>
      </p>
    </section>
  );
};

export default Quote;
