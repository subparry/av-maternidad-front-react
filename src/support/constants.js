export const APP_NAME = "Andrea lactancia";
export const API_BASE_URL = "http://localhost:3000";
