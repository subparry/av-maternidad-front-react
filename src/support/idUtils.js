class idUtils {
  constructor() {
    this.counter = 1;
  }

  createUniqueId(string = "generated-id") {
    const uniqueId = `${string}-${this.counter}`;
    this.counter++;
    return uniqueId;
  }
}

const idUtilsInstance = new idUtils();

export default idUtilsInstance;
