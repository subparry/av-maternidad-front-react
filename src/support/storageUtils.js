class StorageUtils {
  constructor() {
    this.notifiers = [];
  }

  subscribe({ id, callback }) {
    if (this.notifiers.some((notifier) => notifier.id === id)) {
      throw new Error("Already subscribed an observer with id = " + id);
    }
    this.notifiers.push({ id, callback });
  }

  unsubscribe(id) {
    this.notifiers = this.notifiers.filter((notifier) => notifier.id !== id);
  }

  setLocal(key, value) {
    const prevValue = window.localStorage.getItem(key);
    window.localStorage.setItem(key, value);
    this.notifiers.forEach(({ callback }) =>
      callback({ method: "setItem", key, value, prevValue })
    );
  }

  getLocal(key) {
    const value = window.localStorage.getItem(key);
    this.notifiers.forEach(({ callback }) =>
      callback({ method: "getItem", key, value })
    );
    return value;
  }
}

const storageHelper = new StorageUtils();
export default storageHelper;
