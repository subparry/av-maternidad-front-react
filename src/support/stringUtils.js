class StringUtils {
  static truncateAt = (str, num) => {
    return str.slice(0, num).concat("...");
  };
}

export default StringUtils;
