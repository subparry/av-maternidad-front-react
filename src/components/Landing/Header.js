import React from "react";
import { APP_NAME } from "../../support/constants";
import "./styles/header.scss";

const Header = () => {
  return (
    <section
      data-testid="header-section"
      className="header-section header-background"
    >
      <h1 className="text-giant text-white normal">{APP_NAME}</h1>
      <h2 className="text-md text-white light italic">
        Una crianza consciente crea adultos sanos...
      </h2>
    </section>
  );
};

export default Header;
