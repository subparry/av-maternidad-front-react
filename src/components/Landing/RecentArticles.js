import React, { useEffect } from "react";
import { API_BASE_URL } from "../../support/constants";
import Slider from "../shared/Slider";
import ArticlePreviewCard from "../shared/ArticlePreviewCard";

const RecentArticles = () => {
  useEffect(() => {
    fetch(`${API_BASE_URL}/recent_posts`).catch((err) => {});
  }, []);
  return (
    <section data-testid="recent-articles-section">
      <h2>Art√≠culos recientes</h2>
      <Slider
        items={[1, 2, 3, 4, 5, 6, 7, 8].map((e) => (
          <ArticlePreviewCard key={e} />
        ))}
        itemWidthInPx={290}
      />
    </section>
  );
};

export default RecentArticles;
