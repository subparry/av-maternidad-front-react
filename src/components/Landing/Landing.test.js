import { render, fireEvent, waitFor, screen } from "@testing-library/react";
import { APP_NAME, API_BASE_URL } from "../../support/constants";

import React from "react";
import Header from "./Header";
import Quote from "./Quote";
import RecentArticles from "./RecentArticles";

describe("<Header />", () => {
  it("Contains h1 with app title", () => {
    const { getByRole } = render(<Header />);
    const headingElement = getByRole("heading", { name: APP_NAME });
    expect(headingElement).toBeInTheDocument();
    expect(headingElement.tagName.toLowerCase()).toBe("h1");
  });

  it("Has full width background image", () => {
    const { getByTestId } = render(<Header />);
    const sectionElement = getByTestId("header-section");
    expect(sectionElement.classList.contains("header-background")).toBe(true);
  });

  it("has h2 subtitle", () => {
    const { getAllByRole } = render(<Header />);
    const headings = getAllByRole("heading");
    expect(headings.some((ele) => ele.tagName.toLowerCase() === "h2")).toBe(
      true
    );
  });

  it("matches snapshot", () => {
    const { container } = render(<Header />);

    expect(container.firstElementChild).toMatchSnapshot();
  });
});

describe("<Quote />", () => {
  it("has some meaningful quote", () => {
    const { container } = render(<Quote />);
    expect(container.querySelector("p").textContent.length).toBeGreaterThan(10);
  });

  it("matches snapshot", () => {
    const { container } = render(<Quote />);
    expect(container.firstElementChild).toMatchSnapshot();
  });
});

describe("<RecentArticles />", () => {
  it("requests recent articles to api", () => {
    const mockFetch = jest.spyOn(window, "fetch");
    render(<RecentArticles />);
    waitFor(() => {
      expect(mockFetch).toHaveBeenCalledTimes(1);
    });

    expect(mockFetch).toHaveBeenCalledWith(
      `${API_BASE_URL}/recent_posts`,
      expect.anything()
    );
  });

  it("renders <Slider /> component", () => {
    const { container } = render(<RecentArticles />);

    expect(
      container.querySelector(".apa-slider-container")
    ).toBeInTheDocument();
  });
});
