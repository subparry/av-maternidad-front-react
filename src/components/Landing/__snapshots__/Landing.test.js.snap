// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> matches snapshot 1`] = `
<section
  class="header-section header-background"
  data-testid="header-section"
>
  <h1
    class="text-giant text-white normal"
  >
    Andrea lactancia
  </h1>
  <h2
    class="text-md text-white light italic"
  >
    Una crianza consciente crea adultos sanos...
  </h2>
</section>
`;

exports[`<Quote /> matches snapshot 1`] = `
<section
  class="quote-section"
  data-testid="quote-section"
>
  <p
    class="parenthood-quote text-md"
  >
    -La funci√≥n de un hijo es vivir su propia vida, no vivir la vida que su padre ansioso cree que debe vivir. 
    <em>
      -A.S. Neill.
    </em>
  </p>
</section>
`;
